### PR TITLE
feat: add .dockerignore

### LINKED ISSUE #2539: Feature: Optimize dockerfile
### 🔖 Feature description

Optimize dockerfile for shorter build time

### 🎤 Pitch

Optimize dockerfile for shorter build time

### ✌️ Solution

- Use `.dockerignore` file to skip irrelevant files and folders.
- Pin image versions

### 🔄️ Alternative

NA

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by rakshitgondwal**
# Description
Added a .dockerignore file to optimize the build time of Dockerfile.

Fixes #2539

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+*.md
+docker-compose.yml
+.git
+.gitattributes
+.gitignore
+.gitbook.yaml
+LICENSE
+DCO
+docs
+developers-guide
```
Comment (pawan-mehta-dt) [inline .dockerignore:10]: Include CHANGELOG, contrib-chart, developers-guide directories too

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#Markdown files
+*.md
+
+#Docker files
+docker-compose.yml
+
+#Git files
+.git
```
Comment (pawan-mehta-dt) [inline .dockerignore:8]: Please remove .git directory too, it's needed for build

Comment (rakshitgondwal) [reply .dockerignore]: Done!

**Comment by rakshitgondwal** [comment]
Hey! @pawan-mehta-dt. So sorry for the tag but any update on this PR?

**Comment by jatin-jangir** [comment]
hello, `rakshitgondwal` 
there are some  more files that can be added 
```
#Markdown files
**/*.md

#Docker files
docker-compose.yml
Dockerfile
DockerfileEA

#Git files
. gitattributes
.gitignore
.gitbook.yaml

#Folders
CHANGELOG/
contrib-chart/
docs/
developers-guide/
sample-docker-templates/
assets/
charts/
manifests/
specs/

#Files
LICENSE
DCO
688
localhost.key
localhost.crt
```

**Pull request closed without merging**